client/src/document/organisms/toc/placement.tsx

```diff
@@ -0,0 +1,149 @@
+import { useCallback, useEffect, useRef } from "react";
+import { useIsServer } from "../../../hooks";
+import {
+  PlacementStatus,
+  usePlacementStatus,
+} from "../../../placement-context";
+import { useUserData } from "../../../user-context";
+import { usePageVisibility } from "../../../writers-homepage/hooks";
+
+interface Timer {
+  timeout: number | null;
+  start: number | null;
+  notVisible?: boolean;
+}
+
+function viewed(
+  pong: PlacementStatus,
+  observer: IntersectionObserver | null = null
+) {
+  navigator?.sendBeacon(
+    `/pong/viewed?code=${encodeURIComponent(pong?.view)}${
+      pong?.fallback
+        ? `&fallback=${encodeURIComponent(pong?.fallback?.view)}`
+        : ""
+    }`
+  );
+  observer?.disconnect();
+}
+
+export function Placement() {
+  const isServer = useIsServer();
+  const user = useUserData();
+  const pong = usePlacementStatus();
+  const isVisible = usePageVisibility();
+
+  const observer = useRef<IntersectionObserver | null>(null);
+  const timer = useRef<Timer>({ timeout: null, start: null });
+
+  const place = useCallback(
```

Comment:
Why do we `useCallback()`. Couldn't we merge this with the `useEffect()` below that does the disconnect?